### ISSUE TITLE
Stuck with "Processing Photo Metadata" without showing a thing or finish this step

### ISSUE THREAD

**Issue by fabiscafe** [blissd/fotema#104]
After I start Fotema, I see this and it wont change ever:
![grafik](https://github.com/user-attachments/assets/6bf5e4fa-0c86-4335-bac8-b372c2728901)

    LANG=C flatpak run --env=RUST_LOG=fotema=DEBUG,glycin=DEBUG --env=RUST_BACKTRACE=1 app.fotema.Fotema

```
20:50:43.006  INFO fotema::app: XDG_PICTURES_DIR is "/home/fabian/Bilder"
20:50:43.011  INFO fotema::languages: Requested languages: []
20:50:43.011  INFO fotema::languages: Requested languages with fallbacks: [LanguageIdentifier { language: Language(Some("en")), script: None, region: Some(Region("US")), variants: None }]

(fotema:2): Adwaita-WARNING **: 20:50:43.116: AdwNavigationPage 0x64416da78460 is missing a title. To hide a header bar title, consider using AdwHeaderBar:show-title instead.

(fotema:2): Adwaita-WARNING **: 20:50:43.117: AdwNavigationPage 0x64416db26450 is missing a title. To hide a header bar title, consider using AdwHeaderBar:show-title instead.
20:50:43.117  INFO fotema::app::background::bootstrap: Start
20:50:43.117  INFO fotema::app::background::photo_scan: Scanning file system for pictures...
20:50:43.119  INFO fotema::app::background::bootstrap: Scan photos started
20:50:43.273  INFO fotema::app::background::photo_scan: Found 13882 photos to add to database
20:50:43.430  INFO fotema::app::components::albums::album: All view is active so refreshing
20:50:43.430  INFO fotema::app::background::bootstrap: Scan photos completed
20:50:43.430  INFO fotema::app::background::video_scan: Scanning file system for videos...
20:50:43.525  INFO fotema::app::background::video_scan: Found 10 videos to add to database
20:50:43.790  INFO fotema::app::components::albums::months_album: Month is inactive so clearing
20:50:43.790  INFO fotema::app::components::albums::years_album: Year is inactive so clearing
20:50:43.790  INFO fotema::app::components::albums::album: Selfies view is inactive so clearing
20:50:43.790  INFO fotema::app::components::albums::album: Animated view is inactive so clearing
20:50:43.790  INFO fotema::app::components::albums::album: Videos view is inactive so clearing
20:50:43.790  INFO fotema::app::components::albums::places_album: Places view is inactive so clearing
20:50:43.790  INFO fotema::app::components::albums::folders_album: Folders view is inactive so clearing
20:50:43.790  INFO fotema::app::components::albums::album: Folder view is inactive so clearing
20:50:43.802  INFO fotema::app::background::bootstrap: Scan videos started
20:50:43.802  INFO fotema::app::background::bootstrap: Scan videos completed
20:50:43.802  INFO fotema::app::background::photo_enrich: Enriching photos...
20:50:43.839  INFO fotema::app::background::photo_enrich: Found 13882 photos as candidates for enriching
20:50:43.900  INFO fotema::app::background::bootstrap: Photo enrichment started
20:50:44.263 ERROR fotema::app::background::photo_enrich: Failed to update previews: NOT NULL constraint failed: pictures_geo.longitude
```

I already tried to uninstall and delete app-data, but without any success. This is with the official flathub-flatpak.

**Comment by blissd** [comment]
Hi @fabiscafe, thank you for your bug report. Frustratingly, the logging output isn't too helpful here. I would guess that a picture has a latitude, but no sibling longitude in its metadata. However, I think Fotema should handle that. I'll take a harder look and see if I can see how this situation arises.

**Comment by fabiscafe** [comment]
Hi and thanks! Is there something I could do to look up what file it is? So I could send you.

**Comment by blissd** [comment]
Hmm... I think it might be that the longitude is computed as "not-a-number", which [SQLite treats as null](https://stackoverflow.com/questions/43526800/sqliteconstraintexception-not-null-constraint-failed-on-a-real-field). I'll add a NAN check and cut a new release soon.

**Comment by blissd** [comment]
> Hi and thanks! Is there something I could do to look up what file it is? So I could send you.

I don't think there is. I extract the metadata from every photo and then save them in the database in one go, instead of one by one. My reason for that is it many times faster... however, I see now that the trade off is it is a pain to debug when there is a problem!

Perhaps in a future release I should allow a command line option to not batch up the metadata for use when debugging.

**Comment by blissd** [comment]
I've published version v1.9.5 with the fix above so it should be available on Flathub later today. Please let me know if it fixes the bug you are experiencing.

**Comment by blissd** [comment]
Hey, I thought I recognized your handle! Thanks for the nice comment about Fotema on Mastodon.

**Comment by fabiscafe** [comment]
> I've published version v1.9.5 with the fix above so it should be available on Flathub later today. Please let me know if it fixes the bug you are experiencing.

Yes, that completely fixes it! Wow thanks. 

> Hey, I thought I recognized your handle! Thanks for the nice comment about Fotema on Mastodon.

You're welcome. A proper gallery is such a good thing that was just missing on linux. I have to thank you!

----

Fixed with: https://github.com/blissd/fotema/commit/a15f18fdf51566b44d525ff35630e94d14c31299
